sdk/api_fingerprint.go

```diff
@@ -11,328 +11,118 @@ package sdk
-	RequestId     optional.String
-	LinkedId      optional.String
-	Limit         optional.Int32
-	PaginationKey optional.String
-	Before        optional.Int64
+type requestDefinition struct {
+	StatusCodeResultsFactoryMap map[int]func() interface{}
+	Path                        func(params ...string) string
 }
 
-func (a *FingerprintApiService) GetVisits(ctx context.Context, visitorId string, localVarOptionals *FingerprintApiGetVisitsOpts) (Response, *http.Response, error) {
-	var (
-		localVarHttpMethod  = http.MethodGet
-		localVarPostBody    interface{}
-		localVarFileName    string
-		localVarFileBytes   []byte
-		localVarReturnValue Response
-	)
-
-	// create path and map variables
-	localVarPath := a.client.cfg.basePath + "/visitors/{visitor_id}"
-	localVarPath = strings.Replace(localVarPath, "{"+"visitor_id"+"}", fmt.Sprintf("%v", visitorId), -1)
-
-	localVarHeaderParams := make(map[string]string)
-	localVarQueryParams := url.Values{}
-	localVarFormParams := url.Values{}
+func createDeleteVisitorDataDefinition() requestDefinition {
+	return requestDefinition{
+		Path: func(args ...string) string {
+			pathParams := []string{"visitor_id"}
 
-	localVarQueryParams.Add("ii", "fingerprint-pro-server-go-sdk/5.0.2")
-	if localVarOptionals != nil {
+			path := "/visitors/{visitor_id}"
 
-		if localVarOptionals.RequestId.IsSet() {
-			localVarQueryParams.Add("request_id", parameterToString(localVarOptionals.RequestId.Value(), ""))
-		}
-		if localVarOptionals.LinkedId.IsSet() {
-			localVarQueryParams.Add("linked_id", parameterToString(localVarOptionals.LinkedId.Value(), ""))
-		}
-		if localVarOptionals.Limit.IsSet() {
-			localVarQueryParams.Add("limit", parameterToString(localVarOptionals.Limit.Value(), ""))
-		}
-		if localVarOptionals.PaginationKey.IsSet() {
-			localVarQueryParams.Add("paginationKey", parameterToString(localVarOptionals.PaginationKey.Value(), ""))
-		}
-		if localVarOptionals.Before.IsSet() {
-			localVarQueryParams.Add("before", parameterToString(localVarOptionals.Before.Value(), ""))
-		}
-	}
-	// to determine the Content-Type header
-	localVarHttpContentTypes := []string{}
-
-	// set Content-Type header
-	localVarHttpContentType := selectHeaderContentType(localVarHttpContentTypes)
-	if localVarHttpContentType != "" {
-		localVarHeaderParams["Content-Type"] = localVarHttpContentType
-	}
-
-	// to determine the Accept header
-	localVarHttpHeaderAccepts := []string{"application/json"}
-
-	// set Accept header
-	localVarHttpHeaderAccept := selectHeaderAccept(localVarHttpHeaderAccepts)
-	if localVarHttpHeaderAccept != "" {
-		localVarHeaderParams["Accept"] = localVarHttpHeaderAccept
-	}
-	if ctx != nil {
-		// API Key Authentication
-		if auth, ok := ctx.Value(ContextAPIKey).(APIKey); ok {
-			var key string
-			if auth.Prefix != "" {
-				key = auth.Prefix + " " + auth.Key
-			} else {
-				key = auth.Key
+			for i, arg := range args {
+				path = strings.Replace(path, "{"+pathParams[i]+"}", arg, -1)
```

Comment:
This code is a little hard to read. We got 3 instances of this code block, all has `pathParams` array with length 1. Isn't it better to just append the arg to the path string, like `path := "/visitors/" + args[0]`? What is the advantage of creating an array and looping over it? It surely loses simplicity.

Reply:
Generally the idea here is to support multiple path paremeters. For now it might look a bit "ugly", but this will reduce one manual step that we would have to make in case if we were to introduce such endpoint.

Reply:
How does it reduce the manual steps if we were to introduce an endpoint with multiple path parameters? Instead of changing one line of code, in your case we would change two lines of code, `pathParams` and `path` parts. And it will still be more lines of code in total. Not to mention this code is repeated. If we were to add more endpoints, we would need to copy paste some # of lines to the new endpoint, and if we were to change this implementation, we would make the same changes # of endpoints times.

Reply:
We won't need to do it - this code is autogenerated by swagger 🙂. But I will look for a ways to make it less repeatable, while keeping it autogenerated.

Reply:
Sorry, I forgot to mention this in my initial response. This code is generated by swagger. The reason why `pathParams` is an array is so that it can store the URL parameters as they are defined in our OpenAPI schema. In case if there were more parameters in URL, e.g: `{request_id}/{visitor_id}` it would work with these as well.